src/Factory/NavigationItemFactory.php

```diff
@@ -135,4 +150,17 @@ private function addMatchesConfig(array &$config, NavigationItem $item)
             $config[HydratorItem::OPTION_MATCHES][] = $this->getFactory()->flattenProperty($match);
         }
     }
+
+    /**
+     * @param Item $item
+     * @return bool
+     */
+    private function filterOut(Item $item): bool
+    {
+        if (!$this->filter) {
+            return false;
+        }
+
+        return $this->filter->filterOut($item);
```

Comment:
You don't need to use my version - it's just "shorter" :-) I like your explicit version more though. 

```php
return $this->filter ? $this->filter->filterOut($item) : null;
```